src/GNDStk/v1.9/containers/Axes/src/custom.hpp

```diff
@@ -0,0 +1,30 @@
+private:
+
+  static inline helpMap help = {};
+
+  /**
+   *  @brief Custom construct function
+   *
+   +  Sort the elements by index so that the indices are in order
+   */
+  void construct() {
+
+    // useful lambdas
+    auto getIndex = [] ( const auto& variant )
+                       { return variant.index().value(); };
+
+    std::sort( this->content.choice.begin(), this->content.choice.end(),
+               [&] ( const auto& left, const auto& right )
+                   { return std::visit( getIndex, left ) <
+                            std::visit( getIndex, right ); } );
+  }
+
+public:
+
+  /**
+   *  @brief Return the axis and grid elements defined in the axes element
+   */
+  decltype(auto) size() const {
```

Comment:
auto should be sufficient, as it's just going to be a std::size_t. By the way, it might make sense to have the autogenerator code make size() functions automatically, except it's presumably possible that a class it creates could have more than one vector in it, in which case we wouldn't know which one size() is for. So doing those in the customization seems like the way to go.

Reply:
In this case yes, auto would have been sufficient since it is an integral type. But I've been bitten before by inadvertently copying data by using auto as the return type instead of decltype(auto).

And again, this might indeed be a candidate for autogeneration.